searchlib/src/vespa/searchlib/query/query_term_simple.cpp

```diff
@@ -4,19 +4,66 @@
 #include "base.h"
 #include <vespa/vespalib/objects/visit.h>
 #include <vespa/vespalib/util/classname.h>
-#include <vespa/vespalib/locale/c.h>
 #include <cmath>
 #include <limits>
 #include <charconv>
 
 namespace {
 
 template <typename N>
-bool isValidInteger(int64_t value)
+constexpr bool isValidInteger(int64_t value) noexcept
 {
-    return value >= std::numeric_limits<N>::min() && value <= std::numeric_limits<N>::max();
+    return (value >= std::numeric_limits<N>::min()) &&
+           (value <= std::numeric_limits<N>::max());
 }
 
+constexpr bool isRepresentableByInt64(double d) noexcept {
+    return (d > double(std::numeric_limits<int64_t>::min())) &&
+           (d < double(std::numeric_limits<int64_t>::max()));
+}
+
+bool isFullRange(vespalib::stringref s) noexcept {
+    const size_t sz(s.size());
+    return (sz >= 3u) &&
+           (s[0] == '<' || s[0] == '[') &&
+           (s[sz-1] == '>' || s[sz-1] == ']');
+}
+
+struct IntDecoder {
+    static int64_t fromstr(const char * q, const char * qend, const char ** end) noexcept {
+        int64_t v(0);
+        for (;q < qend && isspace(*q); q++);
+        std::from_chars_result err = std::from_chars(q, qend, v, 10);
+        if (err.ec == std::errc::result_out_of_range) {
+            v = (*q == '-') ? std::numeric_limits<int64_t>::min() : std::numeric_limits<int64_t>::max();
+        }
+        *end = err.ptr;
+        return v;
+    }
+    static int64_t nearestDownwd(int64_t n, int64_t min) noexcept { return (n > min ? n - 1 : n); }
+    static int64_t nearestUpward(int64_t n, int64_t max) noexcept { return (n < max ? n + 1 : n); }
+};
+
+template <typename T>
+struct FloatDecoder {
+    static T fromstr(const char * q, const char * qend, const char ** end) noexcept {
+        T v(0);
+        for (;q < qend && isspace(*q); q++);
+        std::from_chars_result err = std::from_chars(q, qend, v);
```

Comment:
Additionally, do we support hex floating point values? If so, it looks like we might have to detect+consume any leading `0x` and invoke `from_chars` with [std::chars_format::hex](https://en.cppreference.com/w/cpp/utility/chars_format)

Reply:
We do not support hexencoded integers, so we should not support hex encoded floats either.